packages/angular/ssr/node/src/request.ts

```diff
@@ -83,18 +83,42 @@ function createRequestUrl(nodeRequest: IncomingMessage | Http2ServerRequest): UR
     originalUrl,
   } = nodeRequest as IncomingMessage & { originalUrl?: string };
   const protocol =
-    headers['x-forwarded-proto'] ?? ('encrypted' in socket && socket.encrypted ? 'https' : 'http');
-  const hostname = headers['x-forwarded-host'] ?? headers.host ?? headers[':authority'];
-  const port = headers['x-forwarded-port'] ?? socket.localPort;
+    getFirstForwardedValue(headers['x-forwarded-proto']) ??
+    ('encrypted' in socket && socket.encrypted ? 'https' : 'http');
+  const hostname =
+    getFirstForwardedValue(headers['x-forwarded-host']) ?? headers.host ?? headers[':authority'];
 
   if (Array.isArray(hostname)) {
     throw new Error('host value cannot be an array.');
   }
 
   let hostnameWithPort = hostname;
-  if (port && !hostname?.includes(':')) {
-    hostnameWithPort += `:${port}`;
+  if (!hostname?.includes(':')) {
+    const port = getFirstForwardedValue(headers['x-forwarded-port']);
+    if (port) {
+      hostnameWithPort += `:${port}`;
+    }
   }
```

Comment:
Question: Do we care about the case where two ports are specified like:

```
X-Forwarded-Host: example.com:1234
X-Forwarded-Port: 4321
```

Seems like here we'll favor `1234`. Is that intentional or are we calling this unspecified behavior?

Reply:
From what I understand the `X-Forwarded-Host` should not have a `port`. Unlike the `host` header.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host

Reply:
Oh I see, this is intended for the `Host: example.com:1234` case. Seems like it will also catch the `X-Forwarded-Host: example.com:1234` case, but I'm fine calling that out of scope if it's not expected for that header.

I was going to point out that "host" as a term typically does not include a port, but since this is a de-facto standard I'm not sure what to expect from `X-Forwarded-Host`.